wallet/walletrpc.c

```diff
@@ -655,6 +655,30 @@ static struct command_result *match_psbt_inputs_to_utxos(struct command *cmd,
 	return NULL;
 }
 
+static struct command_result *match_psbt_outputs_to_wallet(struct wally_psbt *psbt,
+							   struct wallet *w)
+{
+	assert(psbt->tx->num_outputs == psbt->num_outputs);
+	tal_wally_start();
+	for (size_t outndx = 0; outndx < psbt->num_outputs; ++outndx) {
+		u32 index;
+		bool is_p2sh;
+		const u8 *script;
+
+		script = wally_tx_output_get_script(tmpctx,
+						    &psbt->tx->outputs[outndx]);
+		if (!script)
+			continue;
+
+		if (!wallet_can_spend(w, script, &index, &is_p2sh))
```

Comment:
this does scanning.  can't we populate the keypath when we create the output in the first place?

Reply:
I don't see how; the psbt is passed in via the JSON RPC interface, may have been crafted manually?

Reply:
ah, right